go-controller/pkg/ovn/pods.go

```diff
@@ -214,16 +214,16 @@ func (oc *Controller) addRoutesGatewayIP(pod *kapi.Pod, podAnnotation *util.PodA
 	return nil
 }
 
-func (oc *Controller) getRoutingExternalGWs(ns string) []net.IP {
+func (oc *Controller) getRoutingExternalGWs(ns string) gatewayInfo {
```

Comment:
this has nothing to do with your code, but we are returning a reference to an object contained within nsInfo here without holding the lock on it?

Reply:
Yup maybe it's safer to return a copy?

Reply:
In theory, it could be still messed up as the object returned contains a slice, which is "kind of" a reference in go. The array backing up the slice is the same unless some operation happens on one of the copies.
It might be that it's never the case (as it was not until now), but I'd prefer to be on the safe side and copy the slice (in a new result object).

Reply:
yeah thats a good point